google/generativeai/files.py

```diff
@@ -97,6 +111,10 @@ def list_files(page_size=100) -> Iterable[file_types.File]:
         yield file_types.File(proto)
 
 
+async def list_files_async(*args, **kwargs):
```

Comment:
I think this will fail because we want an async iterator to be returned here. 

Can you copy the list_files code and add the missing `async`s?